### ISSUE TITLE
Pretrained model for sth-sth v1 and kinetics got lower accuracy

### ISSUE THREAD

**Issue by Andy1621** [mit-han-lab/temporal-shift-module#65]
Hello, thanks for your great job!!!
Today when I tried to fine-tuning your model and test the accuracy of your model, something strange happened.  I used uniform sampling, but it seemed that both the pretrained model for sth-sth v1 and kinetics got lower accuracy.

**sth-sth v1**
```
CUDA_VISIBLE_DEVICES=1 \
python test_models.py somethingv1 \
--weights /data5/ckli/exp/TSM/TSM_something_RGB_resnet50_shift8_blockres_avg_segment8_e45.pth \ 
--test_segments 8 --batch_size 72 --test_crops 1 --gpus 0
somethingv1: 174 classes
=> shift: True, shift_div: 8, shift_place: blockres

    Initializing TSN with base model: resnet50.
    TSN Configurations:
        input_modality:     RGB
        num_segments:       8
        new_length:         1
        consensus_module:   avg
        dropout_ratio:      0.8

=> backbone: resnet50
=> base model: TSM
Adding temporal shift...
=> n_segment per stage: [8, 8, 8, 8]
=> Processing stage with 3 blocks residual
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Processing stage with 4 blocks residual
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Processing stage with 6 blocks residual
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Processing stage with 3 blocks residual
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
video number:11522
DataParallel...
ok
video 72 done, total 72/11522, average 0.133 sec/video, moving Prec@1 33.333 Prec@5 50.000
video 1512 done, total 1512/11522, average 0.018 sec/video, moving Prec@1 21.230 Prec@5 45.767
video 2952 done, total 2952/11522, average 0.015 sec/video, moving Prec@1 20.833 Prec@5 45.088
video 4392 done, total 4392/11522, average 0.014 sec/video, moving Prec@1 20.196 Prec@5 44.627
video 5832 done, total 5832/11522, average 0.014 sec/video, moving Prec@1 20.130 Prec@5 44.153
video 7272 done, total 7272/11522, average 0.013 sec/video, moving Prec@1 20.462 Prec@5 44.417
video 8712 done, total 8712/11522, average 0.013 sec/video, moving Prec@1 20.535 Prec@5 44.284
video 10152 done, total 10152/11522, average 0.013 sec/video, moving Prec@1 20.892 Prec@5 44.464
video 11592 done, total 11592/11522, average 0.013 sec/video, moving Prec@1 20.890 Prec@5 44.393
[0.04477612 0.1875     0.12380952 0.56716418 0.18421053 0.20689655
 0.17164179 0.20588235 0.15533981 0.12621359 0.12       0.06944444
 0.04255319 0.25806452 0.40764331 0.12       0.31325301 0.175
 0.30081301 0.30666667 0.328      0.18965517 0.14814815 0.17647059
 0.         0.16666667 0.13043478 0.         0.03030303 0.14285714
 0.5        0.15384615 0.22222222 0.08791209 0.         0.05555556
 0.13793103 0.26470588 0.         0.03703704 0.38842975 0.3
 0.4        0.36734694 0.19230769 0.45901639 0.05147059 0.22105263
 0.06741573 0.42857143 0.23584906 0.         0.08       0.1
 0.19047619 0.04166667 0.15151515 0.09375    0.         0.58163265
 0.3030303  0.08333333 0.24       0.09090909 0.         0.30555556
 0.05555556 0.03030303 0.16901408 0.30769231 0.22580645 0.03571429
 0.13157895 0.04       0.08333333 0.         0.         0.17647059
 0.04545455 0.56097561 0.05714286 0.14285714 0.1        0.
 0.08571429 0.12068966 0.28378378 0.23809524 0.05555556 0.
 0.40540541 0.17073171 0.04651163 0.32407407 0.48598131 0.06451613
 0.         0.04761905 0.11111111 0.05882353 0.18367347 0.13888889
 0.046875   0.         0.25609756 0.02857143 0.29213483 0.44927536
 0.19298246 0.03296703 0.         0.         0.13513514 0.04761905
 0.50943396 0.         0.10909091 0.38297872 0.02439024 0.29166667
 0.44318182 0.1372549  0.20909091 0.35       0.25       0.42857143
 0.56603774 0.66666667 0.06206897 0.38461538 0.16       0.
 0.         0.06666667 0.33774834 0.0862069  0.         0.03333333
 0.         0.27631579 0.19       0.18604651 0.22727273 0.27966102
 0.36       0.09859155 0.33018868 0.05970149 0.21875    0.176
 0.65454545 0.17741935 0.11666667 0.12962963 0.         0.
 0.06557377 0.02857143 0.24752475 0.03125    0.22222222 0.125
 0.1        0.10344828 0.03921569 0.36363636 0.34951456 0.28301887
 0.40740741 0.27027027 0.15238095 0.07194245 0.10218978 0.44736842]
upper bound: 0.24146238898715408
-----Evaluation is finished------
Class Accuracy 17.87%
Overall Prec@1 20.89% Prec@5 44.39%
```

**kinetics**
```
CUDA_VISIBLE_DEVICES=1 \
python test_models.py kinetics \
--weights /data5/ckli/exp/TSM/TSM_kinetics_RGB_resnet50_shift8_blockres_avg_segment8_e50.pth \
--test_segments 8 --batch_size 72 --test_crops 1 --gpus 0
kinetics: 400 classes
=> shift: True, shift_div: 8, shift_place: blockres

    Initializing TSN with base model: resnet50.
    TSN Configurations:
        input_modality:     RGB
        num_segments:       8
        new_length:         1
        consensus_module:   avg
        dropout_ratio:      0.8

=> backbone: resnet50
=> base model: TSM
Adding temporal shift...
=> n_segment per stage: [8, 8, 8, 8]
=> Processing stage with 3 blocks residual
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Processing stage with 4 blocks residual
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Processing stage with 6 blocks residual
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
=> Processing stage with 3 blocks residual
=> Using fold div: 8
=> Using fold div: 8
=> Using fold div: 8
video number:19754
DataParallel...
ok
video 72 done, total 72/19754, average 0.128 sec/video, moving Prec@1 58.333 Prec@5 77.778
video 1512 done, total 1512/19754, average 0.018 sec/video, moving Prec@1 56.944 Prec@5 80.026
video 2952 done, total 2952/19754, average 0.015 sec/video, moving Prec@1 58.469 Prec@5 81.199
video 4392 done, total 4392/19754, average 0.035 sec/video, moving Prec@1 58.880 Prec@5 81.557
video 5832 done, total 5832/19754, average 0.076 sec/video, moving Prec@1 59.053 Prec@5 81.481
video 7272 done, total 7272/19754, average 0.088 sec/video, moving Prec@1 59.131 Prec@5 81.518
video 8712 done, total 8712/19754, average 0.107 sec/video, moving Prec@1 59.160 Prec@5 81.531
video 10152 done, total 10152/19754, average 0.111 sec/video, moving Prec@1 58.954 Prec@5 81.511
video 11592 done, total 11592/19754, average 0.122 sec/video, moving Prec@1 58.773 Prec@5 81.444
...
```
The process is stopped for it runs more and more slowly, but it really don't work well......
Are there some errors in the `test_model.py` or the models? Hopefully to your reply.

**Comment by tonylins** [comment]
Hi, I think the possible reason could be data processing. How did you process your data?

**Comment by Andy1621** [comment]
I just use the code for data processing in your repo...
